src/react/WithQueries.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { Function1 } from 'fp-ts/lib/function';
+import { declareQueries } from './declareQueries';
+import { QueryResult } from '../QueryResult';
+import {
+  EnforceNonEmptyRecord,
+  ObservableQueries,
+  ProductL,
+  ProductP,
+  ProductA
+} from '../util';
+
+type Params<A> = A extends void
+  ? {}
+  : {
+      params: A;
+    };
+
+type Props<R extends ObservableQueries> = {
+  queries: EnforceNonEmptyRecord<R>;
+  render: Function1<QueryResult<ProductL<R>, ProductP<R>>, React.ReactNode>;
+} & Params<ProductA<R>>;
+
+export function WithQueries<P extends ObservableQueries>(props: Props<P>) {
```

Comment:
One last minor thing: could you add a JSDOC comment here similar to what has been done [here](https://github.com/buildo/avenger/blob/master/src/react/declareQueries.tsx#L42-L49)?

Reply:
totally legit 👍